chaincode/src/fees/settleFeeCreditReceipts.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) Gala Games Inc. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { FeeBalanceCreditReceipt, FeeReceiptStatus, SettleFeeCreditReceiptsResponse } from "@gala-chain/api";
+import { plainToInstance } from "class-transformer";
+
+import { GalaChainContext } from "../types";
+import { getObjectsByKeys, putChainObject } from "../utils";
+
+export interface SettleFeeCreditReceiptsParams {
+  chainKeys: string[];
+}
+
+export async function settleFeeCreditReceipts(
+  ctx: GalaChainContext,
+  data: SettleFeeCreditReceiptsParams
+): Promise<SettleFeeCreditReceiptsResponse> {
+  const results = await getObjectsByKeys(ctx, FeeBalanceCreditReceipt, data.chainKeys);
+
+  for (const receipt of results) {
+    receipt.status = FeeReceiptStatus.Settled;
+
+    await putChainObject(ctx, receipt);
+  }
```

Comment:
I was going to suggest that we can do it in parallel as follows: 

```suggestion
  await Promise.all(results.map(async (receipt) => {
    receipt.status = FeeReceiptStatus.Settled;
    await putChainObject(ctx, receipt);
  }));
```

But from the other hand it is not saved immediately, but instead put to cache - which is somehow synchronous (all execution in the same thread). So ultimately it doesn't matter 🤔 And it won't speed up anything.

Reply:
Yeah, `Promise.all()` bit me once in the past, I forget exactly where it was in our assets code, but I had to switch it out to a loop. Ever since I find it a lot simpler to read and reason about when it's just in a loop like this.